<launch>
  <!-- get logs in gui -->
  <!--<node pkg="rqt_console" exec="rqt_console" name="rqt_console" output="log"/>-->
	<node pkg="tello_driver" exec="tello_driver_main" name="tello_driver_main" namespace="tello" output="log">
    <!--<param name="drone_ip" value="127.0.0.1" />  for testing-->
  </node>
  <!-- drone manual control -->
  <node pkg="tello_ros" exec="middleware_joy_node" name="middleware_joy_node" namespace="middleware" output="log">
    <remap from="output_cmd_vel" to="/tello/cmd_vel"/>
    <remap from="output_tello_action" to="/tello/tello_action"/>
  </node>
  <node pkg="tello_driver" exec="tello_joy_main" name="tello_joy_main" namespace="input" output="screen">
    <remap from="cmd_vel" to="/middleware/input_cmd_vel"/>
    <remap from="tello_action" to="/middleware/input_tello_action"/>
  </node>
  <node pkg="joy" exec="joy_node" name="tello_joy_main" namespace="input" output="screen"/>
  <!-- end of drone manual control -->
  <node pkg="tello_ros" exec="marker_follower" name="follower" namespace="solution" output="log">
    <remap from="out_cmd_vel" to="/tello/cmd_vel"/>
    <remap from="srv_tello_action" to="/tello/tello_action"/>
    <remap from="action_response" to="/tello/tello_response"/>
    <remap from="input_images" to="/tello/image_raw"/>
    <remap from="flight_status" to="/tello/flight_data"/>
  </node>
  <node pkg="image_view" exec="image_view" name="image_view" namespace="view" output="screen">
    <remap from="image" to="/solution/output_images"/>
  </node>
</launch>
